src/redux/reducers/questions/questionsDataReducer.js

```diff
@@ -0,0 +1,112 @@
+import { ACTION_TYPES } from "../../actions";
+import { RESOURCE_TYPES } from "../../dispatchers";
+
+export const questionsData = function (
+  state = {
+    // boolean flag to indicate whether a request is in progress
+    isFetching: false,
+    // boolean flag to indicate if a request has failed
+    fetchFailed: false,
+    // reason for failure of request
+    fetchFailedReason: "",
+    // object containing data of why the request failed
+    fetchFailedObj: {},
+    // map of questions data
+    questionsMap: {},
+    // map of french questions data
+    questionsMapFr: {},
+  },
+  action
+) {
+  switch (action.type) {
+    case ACTION_TYPES.NETWORK_REQUEST:
+      if (
+        action.resourceType === RESOURCE_TYPES.QUESTIONS ||
+        action.resourceType === RESOURCE_TYPES.QUESTION
+      ) {
+        return {
+          ...state,
+          isFetching: true,
+          fetchFailed: false,
+          fetchFailedReason: "",
+          fetchFailedObj: {},
+        };
+      }
+      break;
+    case ACTION_TYPES.NETWORK_RECEIVED:
+      // Questions EN
+      if (
+        action.resourceType === RESOURCE_TYPES.QUESTIONS ||
+        action.resourceType === RESOURCE_TYPES.QUESTION
+      ) {
+        let data = action.body;
+        if (action.resourceType === RESOURCE_TYPES.QUESTION) {
+          data = [data];
+        }
+        if (Array.isArray(data) && data.length > 0) {
+          let newQuestionsMap = { ...state.questionsMap };
+          data.forEach((value) => {
+            newQuestionsMap[value.id] = {
+              ...newQuestionsMap[value.id],
+              ...value,
+            };
+          });
+
+          return {
+            ...state,
+            isFetching: false,
+            fetchFailed: false,
+            fetchFailedReason: "",
+            fetchFailedObj: {},
+            questionsMap: newQuestionsMap,
+          };
```

Comment:
If the question map is part of the state, is it necessary to return it separately as well?